doc/guest-os-compatibility: document host/guest compat for containers (#15073)
canonical/lxd: doc/guest-os-compatibility.md

```bash
lxc config device remove v1 iso
```

## Containers

> Contrary to virtual machines, container guests rely on the host's kernel for their execution. Since, each Linux distribution comes with a different set of features supported by their official kernels, the possibilities are almost endless.

Reviewer: ```suggestion
Unlike virtual machines, container guests rely on the host's kernel for execution. Since each Linux distribution ships with a unique set of features supported by their official kernels, the possibilities are almost endless. 
```

> As such, the following compatibility table focuses on hosts running Ubuntu LTS releases with LXD `5.21/stable` and running Ubuntu releases as container guests. The main factor to consider for compatibility is the `cgroup` version required by the container and used by the host.

Reviewer: ```suggestion
As such, the following compatibility table focuses on hosts running Ubuntu LTS releases with LXD `5.21/stable` and Ubuntu releases as container guests. The main compatibility factor is the `cgroup` version required by the container and supported by the host.
```
